source/SonicHeroes.WidescreenFix/dllmain.cpp

```diff
@@ -1404,6 +1405,17 @@ void Init()
 		injector::WriteMemory<uint32_t>(ptr_722FA0, 0, true);
 	}
 
+	if (WarpSwitchEffectMode)
+	{
+		uintptr_t loc_41BAD5 = reinterpret_cast<uintptr_t>(hook::pattern("83 7F 50 04 75 0D 6A 02 6A 0A E8 ? ? ? ? 6A 06 EB 0B 6A 05 6A 0A E8 ? ? ? ? 6A 02 6A 0B").get_first(0)) + 0x10;
```

Comment:
I'd suggest to use `auto loc_41BAD5 = hook::pattern("83 7F 50 04 75 0D 6A 02 6A 0A E8 ? ? ? ? 6A 06 EB 0B 6A 05 6A 0A E8 ? ? ? ? 6A 02 6A 0B").get_first(0x10);`

Reply:
I think this is OK. Strongly typing it into a `uintptr_t` makes it easy to manipulate and read because it's clear which type it is and don't have to deal with pattern detector's other functions. I did it this way everywhere else.